### DISCUSSION TITLE
Battery drain

### DISCUSSION THREAD

**Discussion by bharath-so** [official-stockfish/Stockfish#4362] [General]
Hi, I am facing an issue where some of the iOS devices battery gets drained very quickly / device gets heat when using StockFish.
I am here for some suggestions which would help me reduce this issue. Like any configurations on UCI Options or any other way.

**Comment by dubslow** [comment]
Not really anything to be done. Exploring gamespace -- playing good chess -- takes computation, and computation takes battery, it's a physical problem, not a stockfish configuration problem

**Comment by vondele** [comment]
I think there is not much that can be done, except for using a single thread (or few threads) during analysis. Intrinsically, SF is optimized to compute as much as possible in a short time, and has thus high demands for compute and thus power.

**Comment by dubslow** [comment]
Indeed, all you can do is trade strength for power. Using a lot of threads probably reduces the strength-per-power efficiency ratio, but increases the strength-per-time ratio (i.e. you get faster results for equal strength, or stronger results for equal time).

Basically, there's time, battery and strength, and you can only choose two of the three.

**Comment by vdbergh** [comment]
It seems to me that if you run SF on a cellphone and at the same time want to conserve battery then you are not interested in obtaining maximal strength. So you want to throttle SFs CPU usage. It is the task of the OS to make that possible. I don't know if iOS does.
